OutOfSchool/OutOfSchool.WebApi.Tests/Services/DirectionSensitiveServiceTests.cs

```diff
@@ -0,0 +1,171 @@
+    private Mock<ICurrentUserService> currentUserServiceMock;
+    private Mock<IMinistryAdminService> ministryAdminServiceMock;
+    private Mock<IRegionAdminService> regionAdminServiceMock;
+
+    [SetUp]
+    public void SetUp()
+    {
+        mapper = new Mock<IMapper>();
+        var builder =
+            new DbContextOptionsBuilder<OutOfSchoolDbContext>().UseInMemoryDatabase(
+                databaseName: "OutOfSchoolTestDB");
+
+        options = builder.Options;
+        context = new OutOfSchoolDbContext(options);
+
+        repo = new EntityRepositorySoftDeleted<long, Direction>(context);
+        repositoryWorkshop = new WorkshopRepository(context);
+        localizer = new Mock<IStringLocalizer<SharedResource>>();
+        logger = new Mock<ILogger<DirectionService>>();
+        currentUserServiceMock = new Mock<ICurrentUserService>();
+        ministryAdminServiceMock = new Mock<IMinistryAdminService>();
+        regionAdminServiceMock = new Mock<IRegionAdminService>();
+
+        service = new DirectionService(
+            repo,
+            repositoryWorkshop,
+            logger.Object,
+            localizer.Object,
+            mapper.Object,
+            currentUserServiceMock.Object,
+            ministryAdminServiceMock.Object,
+            regionAdminServiceMock.Object);
+
+        SeedDatabase();
+    }
+
+    [Test]
+    [Order(6)]
+    public async Task Update_WhenEntityIsValid_UpdatesExistedEntity()
+    {
+        // Arrange
+        var changedEntity = new DirectionDto()
+        {
+            Id = 1,
+            Title = "ChangedTitle1",
+        };
+
+        var expected = await repo.GetById(changedEntity.Id).ConfigureAwait(false);
+
+        mapper.Setup(m => m.Map<Direction>(changedEntity)).Returns(expected);
+        mapper.Setup(m => m.Map<DirectionDto>(expected)).Returns(changedEntity);
+
+        // Act
+        var result = await service.Update(changedEntity).ConfigureAwait(false);
+
+        // Assert
+        Assert.That(changedEntity.Title, Is.EqualTo(result.Title));
+    }
+
+    [Test]
+    [Order(7)]
+    public void Update_WhenEntityIsInvalid_ThrowsDbUpdateConcurrencyException()
+    {
+        // Arrange
+        var changedEntity = new DirectionDto()
+        {
+            Title = "NewTitle1",
+        };
+        var expected = new Direction()
+        {
+            Title = "NewTitle1",
+        };
+        mapper.Setup(m => m.Map<Direction>(changedEntity)).Returns(expected);
+
+        // Act and Assert
+        Assert.ThrowsAsync<DbUpdateConcurrencyException>(
+            async () => await service.Update(changedEntity).ConfigureAwait(false));
+    }
+
+    private void SeedDatabase()
+    {
+        using var ctx = new OutOfSchoolDbContext(options);
+        {
+            ctx.Database.EnsureDeleted();
+            ctx.Database.EnsureCreated();
+
+            ctx.Institutions.Add(new Institution()
+            {
+                Id = new Guid("af475193-6a1e-4a75-9ba3-439c4300f771"),
```

Comment:
(Readablity) Please make `"af475193-6a1e-4a75-9ba3-439c4300f771"` guid a local variable and use it for `InstitutionId`, as it's not fun to check if all symbols are same everywhere
Also please prepare different local variable with guid for `InstitutionHierarchyId` (At least I believe they should be different for real db).

Reply:
Resolved but nothing done.

Reply:
Sorry Kyrylo, it is my fault. I was wrong (was looking at the `ApplicationSensitiveServiceTests`, but commented here). 
The initial comment is resolved. 
Sorry again.